docs/codecs/gzip/v1.0.rst

```diff
@@ -0,0 +1,120 @@
+========================
+Gzip Codec (version 1.0)
+========================
+-----------------------------
+ Editor's draft 26 July 2019
+-----------------------------
+
+Specification URI:
+    https://purl.org/zarr/spec/codecs/gzip/1.0
+Issue tracking:
+    `GitHub issues <https://github.com/zarr-developers/zarr-specs/labels/codecs-gzip-v1.0>`_
+Suggest an edit for this spec:
+    `GitHub editor <https://github.com/zarr-developers/zarr-specs/blob/core-protocol-v3.0-dev/docs/codecs/gzip/v1.0.rst>`_
+
+Copyright 2019 `Zarr core development
+team <https://github.com/orgs/zarr-developers/teams/core-devs>`_ (@@TODO
+list institutions?). This work is licensed under a `Creative Commons
+Attribution 3.0 Unported
+License <https://creativecommons.org/licenses/by/3.0/>`_.
+
+----
+
+
+Abstract
+========
+
+This specification defines an codec for chunk compression using Gzip
+
+
+Status of this document
+=======================
+
+This document is a **Work in Progress**. It may be updated, replaced
+or obsoleted by other documents at any time. It is inappapropriate to
+cite this document as other than work in progress.
+
+Comments, questions or contributions to this document are very
+welcome. Comments and questions should be raised via `GitHub issues
+<https://github.com/zarr-developers/zarr-specs/labels/codecs-gzip-v1.0>`_. When
+raising an issue, please add the label "codecs-gzip-v1.0".
+
+This document was produced by the `Zarr core development team
+<https://github.com/orgs/zarr-developers/teams/core-devs>`_.
+
+
+Document conventions
+====================
+
+Conformance requirements are expressed with a combination of
+descriptive assertions and [RFC2119]_ terminology. The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative
+parts of this document are to be interpreted as described in
+[RFC2119]_. However, for readability, these words do not appear in all
+uppercase letters in this specification.
+
+All of the text of this specification is normative except sections
+explicitly marked as non-normative, examples, and notes. Examples in
+this specification are introduced with the words "for example".
+
+
+Chunk encoding/decoding with Gzip
+=================================
+
+@@TODO define how chunks are encoded and decoded
+
+Chunks are encoded and decoded using the Gzip compression algorithm defined in
+[RFC1951]_ and [RFC1952]_. The compression level is an integer from 0 to 9
```

Comment:
I don't fully understand all the details here, but this spec will need to be clear I think about the fact that the encoded data should conform to the gzip file format, i.e., include the header and trailer, and not be just a DEFLATE data stream. 

To put it another way, currently in numcodecs we have both a zlib and a gzip codec, which are different. (It was my error originally to think these could be the same.) This spec should match the behaviour of the numcodecs gzip codec.

Reply:
FWIW I think it would be OK to merge this PR without fully working out how to pin down exactly all of the details. But I think it would be worth adding some clarifying statement here just to make sure we have some sense of the distinction.

Reply:
Thanks for the feedback on this @alimanfoo. I've updated the text to be a little more clear about conforming to the gzip file format and added another `@@TODO` for that specific point